enclave/attest_plugin.c

```diff
@@ -105,30 +106,46 @@ oe_result_t oe_get_evidence(
     OE_CHECK(plugin->get_evidence(
         plugin,
         custom_claims,
-        custom_claims_length,
+        custom_claims_size,
         optional_parameters,
         optional_parameters_size,
         &plugin_evidence,
         &plugin_evidence_size,
         endorsements_buffer ? &plugin_endorsements : NULL,
         endorsements_buffer ? &plugin_endorsements_size : NULL));
 
-    // Wrap the attestation header around the evidence.
-    OE_CHECK(_wrap_with_header(
-        format_id,
-        plugin_evidence,
-        plugin_evidence_size,
-        &total_evidence_buf,
-        &total_evidence_size));
-
-    if (endorsements_buffer)
+    // When requested, wrap the attestation header around the evidence.
+    if (wrap_with_header)
     {
         OE_CHECK(_wrap_with_header(
             format_id,
-            plugin_endorsements,
-            plugin_endorsements_size,
-            &total_endorsements_buf,
-            &total_endorsements_size));
+            plugin_evidence,
+            plugin_evidence_size,
+            &total_evidence_buf,
+            &total_evidence_size));
+
+        if (endorsements_buffer)
```

Comment:
It is unnecessary for the endorsements be wrapped around oe_attestation_header_t. 

oe_get_sgx_endorsements() and oe_parse_sgx_endorsements() both deal with raw endorsements without wrapping.

Reply:
Please see above comment.

Reply:
Opened issue 3241 to track the issue.